mlflow/types/type_hints.py

```diff
@@ -300,73 +317,66 @@ def _infer_schema_from_type_hint(type_hint: type[Any]) -> Schema:
     col_spec_type = _infer_colspec_type_from_type_hint(type_hint)
     # Creating Schema with unnamed optional inputs is not supported
     if col_spec_type.required is False:
-        raise InvalidTypeHintException(
-            message=(
-                "To define Optional inputs, use a Pydantic-based type hint definition. See "
-                "https://docs.pydantic.dev/latest/api/base_model/ for pydantic BaseModel examples."
-            )
-        )
+        raise InvalidTypeHintException(message=OPTIONAL_INPUT_MSG)
     return Schema([ColSpec(type=col_spec_type.dtype, required=col_spec_type.required)])
 
 
-def _validate_example_against_type_hint(example: Any, type_hint: type[Any]) -> Any:
+def _validate_data_against_type_hint(data: Any, type_hint: type[Any]) -> Any:
     """
-    Validate the example against provided type hint.
+    Validate the data against provided type hint.
     The allowed conversions are:
-        dictionary example with Pydantic model type hint -> Pydantic model instance
+        dictionary data with Pydantic model type hint -> Pydantic model instance
 
     Args:
-        example: The example to validate
+        data: The data to validate
         type_hint: The type hint to validate against
     """
     if _is_pydantic_type_hint(type_hint):
-        # if example is a pydantic model instance, convert it to a dictionary for validation
-        if isinstance(example, pydantic.BaseModel):
-            example_dict = example.dict() if PYDANTIC_V1_OR_OLDER else example.model_dump()
-        elif isinstance(example, dict):
-            example_dict = example
+        # if data is a pydantic model instance, convert it to a dictionary for validation
+        if isinstance(data, pydantic.BaseModel):
+            data_dict = data.dict() if PYDANTIC_V1_OR_OLDER else data.model_dump()
+        elif isinstance(data, dict):
+            data_dict = data
         else:
             raise MlflowException.invalid_parameter_value(
                 "Expecting example to be a dictionary or pydantic model instance for "
-                f"Pydantic type hint, got {type(example)}"
+                f"Pydantic type hint, got {type(data)}"
             )
         try:
-            model_validate(type_hint, example_dict)
+            model_validate(type_hint, data_dict)
         except pydantic.ValidationError as e:
             raise MlflowException.invalid_parameter_value(
-                message=f"Input example is not valid for Pydantic model `{type_hint.__name__}`",
+                message=f"Data doesn't match type hint, error: {e}",
```

Comment:
Can we be more specific when referring to 'data' here? Which data? Let's ensure that we're specifying the expected data structure that the type hint specifies and what they passed in with the link to the docs page for type hint signatures.